src/Discovery/CollaboraDiscoveryFetcher.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+declare(strict_types=1);
+
+namespace Drupal\collabora_online\Discovery;
+
+use Drupal\collabora_online\Exception\CollaboraNotAvailableException;
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Config\ImmutableConfig;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\RequestOptions;
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Symfony\Component\ErrorHandler\ErrorHandler;
+
+/**
+ * Creates a discovery value object.
+ */
+class CollaboraDiscoveryFetcher implements CollaboraDiscoveryFetcherInterface {
```

Comment:
One thing TBD here is the names of the classes.
E.g. instead of `CollaboraDiscoveryFetcher` we could just have `DiscoveryFetcher`. Or we could rely on the namespace and name it `Fetcher`.

To me, the shortname of a class should be sufficiently unique and distinguishable in _most_ places where it is meant to be used, but not necessarily in all.

Reply:
I agree, Collabora is already in the namespace.

Reply:
renamed all of them.